modules/javafx.graphics/src/main/native-glass/mac/GlassViewDelegate.m

```diff
@@ -1044,25 +1059,23 @@ - (void)startDrag:(NSDragOperation)operation
                 if (offset.y > imageHalfY || offset.y < -imageHalfY) {
                     offset.y = imageHalfY * (offset.y > 0 ? 1 : -1);
                 }
-
-                dragPoint.x += offset.x;
-                dragPoint.y -= offset.y;
             }
         }
         else
         {
             // last resource: "empty" image
-            image = [[NSImage alloc] initWithSize:NSMakeSize(1.0f, 1.0f)];
+            rect.size.width = 1.0f;
+            rect.size.height = 1.0f;
+            image = [[NSImage alloc] initWithSize:NSMakeSize(rect.size.width, rect.size.height)];
         }
-        [self->nsView dragImage:image at:dragPoint offset:NSZeroSize event:self->lastEvent pasteboard:pasteboard source:self->nsView slideBack:YES];
 
-        // main thread blocked here until drag completes
+        rect.origin = NSPointToCGPoint(dragPoint);
+        [item setDraggingFrame:rect contents:image];
+    }
 
-        [GlassDragSource setDelegate:nil];
+    self->draggingSource = [[GlassDraggingSource alloc] initWithOperation:operation];
 
-        [image release];
```

Comment:
It looks like you aren't releasing the image now. You need to call `[image release]` to avoid a leak.